src/lib.rs

```diff
@@ -225,9 +227,28 @@ impl Command {
     }
 }
 
+/// Type parameter: SHTC1.
+pub struct ShtC1;
+/// Type parameter: SHTC3.
+pub struct ShtC3;
+/// Type parameter: Generic driver that should work with all SHTCx sensors.
+pub struct ShtGeneric;
+
+/// Marker trait implemented for all supported sensor models.
+pub trait ShtSensor {}
+impl ShtSensor for ShtC1 {}
+impl ShtSensor for ShtC3 {}
+impl ShtSensor for ShtGeneric {}
+
 /// Driver for the SHTCx sensor.
+///
+/// To create an instance of this, use a constructor function like
+/// [`shtc1`](fn.shtc1.html) or [`shtc3`](fn.shtc3.html) depending on your
+/// sensor.
 #[derive(Debug, Default)]
-pub struct ShtCx<I2C, D> {
+pub struct ShtCx<S: ShtSensor, I2C, D> {
+    /// The chosen target sensor.
+    sensor: PhantomData<S>,
     /// The concrete I²C device implementation.
     i2c: I2C,
```

Comment:
Not related to this PR, but is it smart to take ownership of the I2C bus? I mean since it is a bus it should be shared right?

Reply:
So the following does not work:
```diff
diff --git a/examples/linux.rs b/examples/linux.rs
index 5d0a25e..9156779 100644
--- a/examples/linux.rs
+++ b/examples/linux.rs
@@ -8,6 +8,9 @@ fn main() {
     let address = 0x70; // SHTC3
     let mut sht = shtcx::shtc3(dev, address, Delay);
 
+    let address = 0x71; // foo
+    let mut sht2 = shtcx::shtc3(dev, address, Delay);
+

```

Reply:
You can always `destroy()` the driver to get back the instance. But yes, it would be great if there were a better alternative. Since the `i2c` instance is borrowed mutably, it's not that easy though.

Wasn't there some kind of crate for sharing a bus?

Reply:
https://crates.io/crates/shared-bus

Reply:
Oh cool, it seems that no changes need to be done to the driver crates to support `shared-bus` :slightly_smiling_face: We could add an example for multiple sensors.